nixos/modules/services/cluster/k3s/default.nix

```diff
@@ -498,7 +498,7 @@ in
           "network-online.target"
         ];
         wantedBy = [ "multi-user.target" ];
-        path = lib.optional config.boot.zfs.enabled config.boot.zfs.package;
+        path = [ pkgs.nftables ] ++ lib.optional config.boot.zfs.enabled config.boot.zfs.package;
```

Comment:
> ++ lib.optional config.networking.nftables.enable pkgs.nftables

How about only adding `nftable` in case `nftable` is enabled for host?

Reply:
If host doesn't have `nftables` enabled, how will `nftables` work?

> even i'm using iptables mode

In case, you have explicitly chosen `iptables mode` and K3s still requires `nftables` isn't that a bug somewhere else?

> so it is not detectable in nix configuration.

It should be. Same way there is a `extraKubeletConfig`, `extraKubeProxyConfig` is possible.
Likely would be convenient to have a Nix flag to toggle between iptables/nftables matching the host configuration by default.

Reply:
> In case, you have explicitly chosen iptables mode and K3s still requires nftables isn't that a bug somewhere else?

Hmm, i searched the kubernetes code, and found it's just an intended behavior, and it's ok with this error. https://github.com/kubernetes/kubernetes/blob/810e9e212ec5372d16b655f57b9231d8654a2179/cmd/kube-proxy/app/server_linux.go#L533

> It should be. Same way there is a extraKubeletConfig, extraKubeProxyConfig is possible.
Likely would be convenient to have a Nix flag to toggle between iptables/nftables matching the host configuration by default.

You're right, i will change code later.

Reply:
> Hmm, i searched the kubernetes code, and found it's just an intended behavior, and it's ok with this error. https://github.com/kubernetes/kubernetes/blob/810e9e212ec5372d16b655f57b9231d8654a2179/cmd/kube-proxy/app/server_linux.go#L533

I still think the nftables should be added directly, not optional. For example, if i switch the kube-proxy mode from `nftable` to `iptables` or `ipvs` , the k3s also need the `nft` to clean the previous nftables rules.

Reply:
Okay, add `nftables` by default then.